### DISCUSSION TITLE
Single ServiceMonitor for multiple services or one per each service

### DISCUSSION THREAD

**Discussion by prashantkalkar** [prometheus-operator/prometheus-operator#5179] [Q&A]
Hi,

When using prometheus operator, what is the recommended way to create service monitor for application services deployed?

Can a single Service manager with appropriate label selector to select the services for monitoring be used? That way application services just need to add the required label to enable the monitoring. (Assuming that other configuration is same, like port and path).

If the service wants a different configuration for service monitoring then instead of using the default service monitor a separate serviceMonitor resource can be defined.

Does the above approach has any downsides? Or does the service monitors should be defined per service?

Regards
Prashant

**Comment by simonpasquier** [comment]
In general people tend to define  1 service monitor per service because this is what offers the greatest flexibility. Now if your use case can be addressed with a single service monitor, I guess that it's also fine.
